crates/ra_lsp_server/src/lib.rs

```diff
@@ -22,6 +22,7 @@ macro_rules! print {
 }
 
 mod caps;
+mod cargo_check;
```

Comment:
I think it might make sense to extract this into a separate crate. This doesn't really need anything from `ra_ide` API, and a separate crate should be easier to test.